### PR TITLE
Issue/afp 362 pipeline migration generate website info

### PR DESCRIPTION
**Pull request by dpisani-atl**
This PR adds then generate-website-info pipeline stage into the brisk website. This stage receives information about the target project and then creates data structures representing a spec of what should be created in the website.

The main output of the stage is a list of page specs that the next stage can use to write out .js files. Additionally, a sitemap and meta structure is created which contain data used by the pages themselves (this is what goes in pages-list.json and the like). These have been kept in their original format for backwards compatibility.

Apologies about the large PR. Brisk does a lot of stuff it turns out!

### CODE REVIEW

File: packages/website/src/pipeline/stages/generate-website-info/generate-docs-info.test.ts [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+    expect(pages.docsPages).toHaveLength(2);
+    expect(pages.docsPages[0].websitePath).toEqual('website/docs/doc-1/doc-2');
+    expect(pages.docsPages[1].websitePath).toEqual('website/docs/doc-1/doc-3');
+
+    const rootPage = sitemap[0];
+    expect(rootPage.children).toHaveLength(2);
+
+    expect(rootPage.pagePath).toEqual('website/docs/doc-1');
+    // @ts-ignore children may be undefined
```
[the remark this answers is not in the document]
Comment (dpisani-atl) [reply packages/website/src/pipeline/stages/generate-website-info/generate-docs-info.test.ts]: TIL about `!`

File: packages/website/src/pipeline/stages/generate-website-info/index.ts [hunk 2]
```diff
@@ -1,15 +1,119 @@
...
-// Boilerplate, uncomment when used
-// interface StageConfig {}
+interface Sitemap {
+  packages: PackageSitemap[];
+  docs: ProjectDocsSitemap;
+  // data used by the readme nav
+  readme?: {
+    id: string;
+    pagePath: string;
+  }[];
```
[the remark this answers is not in the document]
Comment (dpisani-atl) [reply packages/website/src/pipeline/stages/generate-website-info/index.ts]: it's actually 'info needed by the readme', I used the property name as it was in the old code, but I'll just make the name better here

File: packages/website/src/pipeline/stages/generate-website-info/generate-docs-info.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        !isDocsGroup(docsNode) &&
+        path.basename(docsNode.markdownPath) === 'readme.md';
+
+      const readme = doc.children.find(isReadme);
+      const childDocs = doc.children.filter(c => !isReadme(c));
+
+      if (readme) {
+        docsPages.push({
+          websitePath,
+          markdownPath: (readme as DocsItem).markdownPath,
```
[the remark this answers is not in the document]
Comment (dpisani-atl) [reply packages/website/src/pipeline/stages/generate-website-info/generate-docs-info.ts]: Unfortunately that didn't work in this case

File: packages/website/src/pipeline/stages/generate-website-info/generate-docs-info.test.ts [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+    const { pages, sitemap } = generateDocsInfo(
+      docs,
+      'website/docs',
+      {},
+      'docs-key',
+    );
+
+    expect(pages.docsPages).toHaveLength(2);
+    expect(pages.docsPages[0].websitePath).toEqual('website/docs/doc-1/doc-2');
```
[the remark this answers is not in the document]
Comment (dpisani-atl) [reply packages/website/src/pipeline/stages/generate-website-info/generate-docs-info.test.ts]: omg I didn't know toMatchObject worked like this on arrays. makes my life so much easier

File: packages/website/src/pipeline/stages/generate-website-info/generate-docs-info.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (ryanbraganza) [reply packages/website/src/pipeline/stages/generate-website-info/generate-docs-info.ts]: ohh, because children is a union type.

**Pull request merged**